File: storage/storage-mysql/src/main/java/io/say/better/storage/mysql/dao/repository/SymbolReadRepository.java
package io.say.better.storage.mysql.dao.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import io.say.better.storage.mysql.domain.entity.Symbol;

public interface SymbolReadRepository extends JpaRepository<Symbol, Long> {
	List<Symbol> findByTitleIn(List<String> symbols);

	List<Symbol> findByTitleStartingWith(String name);
}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/dao/repository/SymbolReadRepository.kt
package io.say.better.storage.mysql.dao.repository

import io.say.better.storage.mysql.domain.entity.Symbol
import org.springframework.data.jpa.repository.JpaRepository

interface SymbolReadRepository : JpaRepository<Symbol, Long> {
    fun findByTitleIn(symbols: List<String>): List<Symbol>

    fun findByTitleStartingWith(name: String): List<Symbol>
}
